src/integrators/type.jl

```diff
@@ -101,8 +101,10 @@ mutable struct ODEIntegrator{algType<:OrdinaryDiffEqAlgorithm,IIP,uType,tType,pT
   saveiter::Int
   saveiter_dense::Int
   cache::CacheType
-  callback_cache::Array{uType}
-  previous_condition::Array{uType}
+  callback_cache::CallbackCacheType
```

Comment:
can we just make a callback cache that each package instantiates? Seems easier than keeping track of a bunch of fields.

Reply:
I don't think so we can do that. All these fields are used in DiffEqBase, so they have to be common. Also they need to be in integrator because of thread safety as pointed out here https://github.com/JuliaDiffEq/DiffEqBase.jl/pull/221#issue-280346404

Reply:
Yes, but there can be a `CallbackCache` that the integrator calls that makes all of these arrays appear.